### ISSUE TITLE
Wrong CID/LAC shown in Database Viewer (0.1.25-alpha-b3 )

### ISSUE THREAD

**Issue by He3556** [SecUpwN/Android-IMSI-Catcher-Detector#203]
**1.** "Database Viewer" > "Cell data" (and also under "Cell Information"):
There is always a CellID = 65554 with LAC = 0 when i start the App. I first saw it a few months ago, so it is not a new issue. After a few hours this cell disappears again (in the View "Cell Information") i have to search if it is still in the local db. I hope it is and we keep all data.

**2.** In the "Database Viewer" > "Location data":
There are a few Cells with CellID = -1 and LAC = 0 
- Solution might be: check the values before saving to db. (only save If CellID != -1) 
I never found CellID=0 as a failure, so i would like to keep this data - in fact a "0" is quite suspicious. 

**3.** I think the label "Neighboring Cells" (in View "Cell Information" and in the Debug Log is really confusing. "Neighboring Cells" is a special term for a list that is send by the BTS to the phone - and we don't have access to this kind of data right now. As far as i can remember, we are currently saving the data from the history - like cells the phone was connected to before. 
This is how we build up this list in the View "Cell Information" Is that right? 
So let's discuss it - we need to get this clear for our users and also for our development.

**Comment by E3V3A** [comment]
@He3556  Maybe the #253 will resolve this? Please see OCID [API](http://wiki.opencellid.org/wiki/API): 

```
CID 	must be in the range of 1 to 65535 for GSM
CID 	must be in the range of 1 to 268435455 for UMTS and LTE
CID/BID 	must be in the range 1 to 65535 for CDMA 
```

**Comment by E3V3A** [comment]
Ok so I think this could also be a timing issue. So we may get these LAC/CID values of "0" or "-1" when:
 1. Before mobile set (radio modem) is properly enabled **or**
 2. before AOS API has received new LAC/CID from baseband. 

So lets summarize what we need to do: 
 * **IF** the LAC/CID are really coming from the network, then we should filter as follows: 
    - For GSM/UMTS/LTE we should only accept a LAC in the range: **[1 - 65533]**
    - For CDMA we should only accept accept a LAC in the range: **[1 - 65534]**

But for CID we should filter as: 
    - For GSM we should only accept accept a CID in the range: **[1, 65534]** ??
    - For UMTS/LTE we should only accept accept a CID in the range: **[1, 268435455]**  (0xFFF FFFF) 
    - For CDMA we should only accept accept a CID in the range: **[1, 65534]**

**Comment by E3V3A** [comment]
https://github.com/mozilla/ichnaea/pull/399

**Comment by E3V3A** [comment]
Please check the latest internal build >26. I think I've fixed this in 7f7774a20123f58654a99c232bdf8ffc40356395 . @He3556  Please re-open this if you're not happy.
